spot_micro_joy/scripts/spotMicroJoystickMove.py

```diff
@@ -0,0 +1,150 @@
+        self._vel_cmd_msg.angular.y = 0
+        self._vel_cmd_msg.angular.z = 0
+
+        self._walk_event_cmd_msg = Bool()
+        self._walk_event_cmd_msg.data = True  # Mostly acts as an event driven action on receipt of a true message
+
+        self._stand_event_cmd_msg = Bool()
+        self._stand_event_cmd_msg.data = True
+
+        self._idle_event_cmd_msg = Bool()
+        self._idle_event_cmd_msg.data = True
+
+        rospy.loginfo("Setting Up the Spot Micro Joystick Control Node...")
+
+        # Set up and title the ros node for this code
+        rospy.init_node('spot_micro_joystick_control')
+
+        # Create publishers for commanding velocity, angle, and robot states
+        self._ros_pub_angle_cmd = rospy.Publisher('/angle_cmd', Vector3, queue_size=1)
+        self._ros_pub_vel_cmd = rospy.Publisher('/cmd_vel', Twist, queue_size=1)
+        self._ros_pub_walk_cmd = rospy.Publisher('/walk_cmd', Bool, queue_size=1)
+        self._ros_pub_stand_cmd = rospy.Publisher('/stand_cmd', Bool, queue_size=1)
+        self._ros_pub_idle_cmd = rospy.Publisher('/idle_cmd', Bool, queue_size=1)
+
+        rospy.loginfo("Joystick control node publishers corrrectly initialized")
+
+    def reset_all_motion_commands_to_zero(self):
+        '''Reset body motion cmd states to zero and publish zero value body motion commands'''
+
+        self._vel_cmd_msg.linear.x = 0
+        self._vel_cmd_msg.linear.y = 0
+        self._vel_cmd_msg.linear.z = 0
+        self._vel_cmd_msg.angular.x = 0
+        self._vel_cmd_msg.angular.y = 0
+        self._vel_cmd_msg.angular.z = 0
+
+        self._ros_pub_vel_cmd.publish(self._vel_cmd_msg)
+
+    def reset_all_angle_commands_to_zero(self):
+        '''Reset angle cmd states to zero and publish them'''
+
+        self._angle_cmd_msg.x = 0
+        self._angle_cmd_msg.y = 0
+        self._angle_cmd_msg.z = 0
+
+        self._ros_pub_angle_cmd.publish(self._angle_cmd_msg)
+
+    def on_joy(self, msg):
+        if msg.buttons[self.BUTTON_GREEN] == 1:
+            self._ros_pub_idle_cmd.publish(self._idle_event_cmd_msg)
+            rospy.loginfo('Idle command issued from joystick.')
+            self.mode = self.MODE_IDLE
+
+        if msg.buttons[self.BUTTON_YELLOW] == 1:
+            self._ros_pub_stand_cmd.publish(self._stand_event_cmd_msg)
+            rospy.loginfo('Stand command issued from joystick.')
+            self.mode = self.MODE_STAND
+
+        if msg.buttons[self.BUTTON_BLUE] == 1:
+            self.reset_all_angle_commands_to_zero()
+            rospy.loginfo('Entering joystick angle command mode.')
+            self.mode = self.MODE_ANGLE
+
+        if msg.buttons[self.BUTTON_RED] == 1:
+            self.reset_all_angle_commands_to_zero()
+            self._ros_pub_walk_cmd.publish(self._walk_event_cmd_msg)
+            rospy.loginfo('Entering joystick walk command mode.')
+            self.mode = self.MODE_WALK
+
+        if self.mode == self.MODE_ANGLE:
```

Comment:
Similar to above, I'd make these if, elif statements so only one mode can be true at a time. The motion controller doesn't (currently) handle angle commands outside of stand mode.

Reply:
And this is another issue. With a joystick it is very easy to fire mode-commands within a very short period of time. This can lead to a desync of modes between this nodes. But i will address this later, didn't checked if i have a topic available for changing this mode where by an event sent by that motion_command. For now, i will change it to if-elif here.